stmt/expression.go

```diff
@@ -57,11 +57,11 @@ func NewExpression(arg interface{}) Expression { // nolint: gocyclo
 
 // Identifier is an identifier.
 type Identifier struct {
-	Identifier string
+	Identifier interface{}
```

Comment:
I think we should create a new interface like `IdentifierExpression` that does the trick of creating a private function, implement by the type `string` (if it's possible) and `Raw`, so we have some safe-guard at compile time.

Reply:
How do you do that without being backward incompatible?

Reply:
By definition, changing the type of `Identifier` from `string` to `interface{}` is backward incompatible.
One could rely on a string comparison of `Identifier.Identifier` or pass that value to a function that accepts a string as parameter.

If you have added this to support ``, maybe we should add it as a type like these ones: 
https://github.com/ulule/loukoum/blob/master/stmt/aggregate.go ?